test/optimizer/selectivity_util_test.cpp

```diff
@@ -0,0 +1,473 @@
+}
+
+// NOLINTNEXTLINE
+TEST_F(SelectivityUtilTests, TestIntegerNotEqual) {
+  // Create a constant value expression to pass to ValueCondition.
+  std::unique_ptr<parser::ConstantValueExpression> const_value_expr_ptr =
+      std::make_unique<parser::ConstantValueExpression>(type::TypeId::INTEGER, execution::sql::Integer(4));
+  // Create a value condition to pass to SelectivityUtil.
+  ValueCondition value_condition(catalog::col_oid_t(3), parser::ExpressionType::COMPARE_NOT_EQUAL,
+                                 std::move(const_value_expr_ptr));
+  // Compute selectivity for col != 4.
+  double res = selectivity_util.ComputeSelectivity(
+      common::ManagedPointer<NewColumnStats<execution::sql::Integer>>(&column_stats_obj_2_), value_condition);
+
+  ASSERT_DOUBLE_EQ(0.6, res);
+
+  // Create a constant value expression to pass to ValueCondition.
+  const_value_expr_ptr =
+      std::make_unique<parser::ConstantValueExpression>(type::TypeId::INTEGER, execution::sql::Integer(0));
+
+  // Create a value condition to pass to SelectivityUtil.
+  value_condition =
+      ValueCondition(catalog::col_oid_t(3), parser::ExpressionType::COMPARE_NOT_EQUAL, std::move(const_value_expr_ptr));
+
+  // Compute selectivity for col != 0.
+  res = selectivity_util.ComputeSelectivity(
+      common::ManagedPointer<NewColumnStats<execution::sql::Integer>>(&column_stats_obj_2_), value_condition);
+
+  ASSERT_DOUBLE_EQ(0.7, res);
+}
+
+// NOLINTNEXTLINE
+TEST_F(SelectivityUtilTests, TestBoolEqual) {
+  // TEST PART 1
+  // Create a constant value expression to pass to ValueCondition.
+  std::unique_ptr<parser::ConstantValueExpression> const_value_expr_ptr =
+      std::make_unique<parser::ConstantValueExpression>(type::TypeId::BOOLEAN, execution::sql::BoolVal(true));
+  // Create a value condition to pass to SelectivityUtil.
+  ValueCondition value_condition(catalog::col_oid_t(5), parser::ExpressionType::COMPARE_EQUAL,
+                                 std::move(const_value_expr_ptr));
+
+  // Compute selectivity for col = true.
+  double res = selectivity_util.ComputeSelectivity(
+      common::ManagedPointer<NewColumnStats<execution::sql::BoolVal>>(&column_stats_obj_5_), value_condition);
+
+  ASSERT_DOUBLE_EQ(0.6, res);
+
+  // TEST PART 2
+  // Create a constant value expression to pass to ValueCondition.
+  const_value_expr_ptr =
+      std::make_unique<parser::ConstantValueExpression>(type::TypeId::BOOLEAN, execution::sql::BoolVal(false));
+  // Create a value condition to pass to SelectivityUtil.
+  value_condition = ValueCondition(catalog::col_oid_t(5), parser::ExpressionType::COMPARE_EQUAL,
+                                 std::move(const_value_expr_ptr));
+
+  // Compute selectivity for col = true.
+  res = selectivity_util.ComputeSelectivity(
+      common::ManagedPointer<NewColumnStats<execution::sql::BoolVal>>(&column_stats_obj_5_), value_condition);
+
+  ASSERT_DOUBLE_EQ(0.2, res);
+}
+}  // namespace noisepage::optimizer
```

Comment:
Probably have a test with nullptr stats. It would have helped you find another error in the code.

Reply:
All my tests have null values. I expected the wrong value in few tests. Fixed it now.

Reply:
No I meant nullptr stats, where you expect the default value.